src/views/mainView.jsx

```diff
@@ -1,25 +1,106 @@
-import {selectLocalPeer, selectIsSomeoneScreenSharing, useHMSStore} from "@100mslive/hms-video-react";
-import { TeacherGridView } from "./teacherGridView";
-import { StudentGridView } from "./studentGridView";
-import {ScreenShareView} from "./screenShareView";
-import {ROLES} from "../common/roles";
+import {
+  selectPeers,
+  selectLocalPeer,
+  selectIsSomeoneScreenSharing,
+  useHMSStore,
+} from "@100mslive/hms-video-react";
+import { ScreenShareView } from "./screenShareView";
+import React, { useContext } from "react";
+
+import { AppContext } from "../store/AppContext";
+import { GridCenterView, GridSidePaneView } from "./components/gridView";
+
+import { App, Layouts } from "../common/layout";
 
 export const ConferenceMainView = ({ isChatOpen, toggleChat }) => {
   const localPeer = useHMSStore(selectLocalPeer);
+  const { maxTileCount } = useContext(AppContext);
   const isSomeoneScreenSharing = useHMSStore(selectIsSomeoneScreenSharing);
+  const peers = useHMSStore(selectPeers);
 
-  if (!localPeer) {  // we don't know the role yet to decide how to render UI
+  if (!localPeer) {
+    // we don't know the role yet to decide how to render UI
     return null;
   }
 
-  const amITeacher = localPeer.role === ROLES.TEACHER;
   let ViewComponent;
 
+  // screen sahring is onn
   if (isSomeoneScreenSharing) {
-    ViewComponent = ScreenShareView;
-  } else {
-    ViewComponent = amITeacher ? TeacherGridView : StudentGridView;
+    ViewComponent = (
+      <ScreenShareView isChatOpen={isChatOpen} toggleChat={toggleChat} />
+    );
+  }
+  // No rule is specified so all peer will be shown in default view
+  else if (!App[localPeer.role]) {
+    ViewComponent = (
+      <GridCenterView
+        peers={peers}
+        maxTileCount={maxTileCount}
+        allowRemoteMute={true}
+        isChatOpen={isChatOpen}
+        toggleChat={toggleChat}
+        hideSidePane={true}
+      />
+    );
+  }
+  // Default view with no sideBar view
+  else if (App[localPeer.role].name === Layouts.DEFAULT) {
+    ViewComponent = (
```

Comment:
this will cause flickering, as a new component is being made for each re render of ConferenceMainView